### ISSUE TITLE
Kundera 2.2 based upon Hibernate 3.6.8, java.lang.NoClassDefFoundError with Hibernate 4.1.7.

### ISSUE THREAD

**Issue by shendriks4** [Impetus/kundera#143]
I'm filling out this issue report per request by Vivek. As noted on the mailing list, I'm trying to get Kundera to work by persisting an entity into MongoDB (v2.2). This entity has a relationship with a parent entity which has already been persisted.

When Kundera check the relationships of the to-be-persisted-entity, It calls upon the class 'AbstractPersistentCollection', which is expected to be in the 'org.hibernate.collection' package (Hibernate v3.6.8). We are however using Hibernate v4.1.7, in which the class is relocated to the 'org.hibernate.collection.internal' package.

In short. Kundera 2.2 is using an older release of Hibernate. Causes issues with projects where newer version of Hibernate is used.

**Comment by Pronimo** [comment]
I see that it's been updated to use version 4.1.7.  But if I understand it, from the entire library of Hibernate the only class Kundera uses is only PersistentCollection, is that right?  Isn't it somewhat an overkill?

**Comment by mevivs** [comment]
well, this will be refactored in 2.3 release and kundera-core will be independent of kundera-rdbms.

-Vivek
